tests/unit/test_application.py

```diff
@@ -484,3 +517,26 @@ def test_run_envvar_invalid(snapcraft_yaml, base, monkeypatch):
         "'SNAPCRAFT_REMOTE_BUILD_STRATEGY'. Valid values are 'disable-fallback' and "
         "'force-fallback'"
     )
+
+
+@pytest.mark.parametrize(
+    ("base", "build_base", "is_known_core24"),
+    [
+        ("core20", None, False),
+        ("core20", "core20", False),
+        ("core20", "devel", False),
+        ("core22", None, False),
+        ("core22", "core22", False),
+        ("core22", "devel", False),
+        ("core24", "core22", True),
```

Comment:
Isn't this still incorrect? A `core22` `build-base` should use the non-craft-app codepath right?

Reply:
Correct.  This is the existing behavior as I reverted all code-flow changes in this PR.

I want to fix the dotnet regression in one PR that I can land as a hotfix.  Changes to code paths should be in a separate PR as part of a minor release.